### PR TITLE
At risk user model

### LINKED ISSUE #32: Set Up At Risk User Model
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [x] New feature (change which adds functionality)
- [ ] Chore
## Issue Summary
:fire: Add User model as well as integrate it into the database <br />

## Resources 
List any resources used when working on the issue (e.g. links to articles, stackoverflow, etc.)
see comments for object sent to front end: https://github.com/track-basket/trackbasket_BE/issues/26
current db config: https://app.lucidchart.com/documents/edit/1622996f-085a-4571-b6b8-5a2befb81f84/0_0?beaconFlowId=59AA3EE2FBC575F8
how to define table models: https://blog.theodo.com/2017/03/developping-a-flask-web-app-with-a-postresql-database-making-all-the-possible-errors/

### PR DESCRIPTION
**Pull request by mronauli**
## Who worked on this

- [ ] Alexis
- [x] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)

## Summary of the change.

🔥Add the at risk user model, resource and get patch and post actions and integrate model into db

This closes #32 #33 #34 

Note any questions or struggles as well.

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/models/at_risk_user.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  state = db.Column(db.String)
+  zipcode = db.Column(db.String)
+  phone_number = db.Column(db.String)
+
+  def json(self):
+    return {'data': { 'id': 'at_risk_user', 'attributes': {'name': self.name, 'at_risk_user_id': self.at_risk_user_id, 'address': self.address, 'city': self.city, 'state': self.state, 'zip code': self.zipcode, 'phone number': self.phone_number} } }
+
+  @classmethod 
+  def find_by_id(cls, id):
+    return cls.query.filter_by(at_risk_user_id=id).first()
```
Comment (mronauli) [inline trackbasket_be/models/at_risk_user.py:20]: moved query into db to make resource a little cleaner and separate query logic

File: trackbasket_be/resources/at_risk_user.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from flask_restful import Resource, reqparse, request 
+from models.at_risk_user import AtRiskUser, db 
+from models.basemodel import BaseModel 
+
+class AtRiskUsers(Resource):
+  
+  def get(self, id): 
+    at_risk_user = AtRiskUser.find_by_id(id)
```
Comment (mronauli) [inline trackbasket_be/resources/at_risk_user.py:8]: call query method from AtRiskUser model

File: trackbasket_be/app.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 from flask_sqlalchemy import SQLAlchemy
 from models.volunteer import db, Volunteer
 from resources.volunteer import Volunteers
+from models.at_risk_user import db, AtRiskUser
+from resources.at_risk_user import AtRiskUsers
```
Comment (adumortier) [inline trackbasket_be/app.py:7]: thank you for following our conventions (plural for resources and singular for model)

File: trackbasket_be/models/at_risk_user.py (hunk 1 above)
Comment (adumortier) [inline trackbasket_be/models/at_risk_user.py:20]: I like this helper method, hopefully we have time to refactor and turn that into a module and use it for other models as well

File: trackbasket_be/resources/at_risk_user.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    if at_risk_user:
+      return at_risk_user.json(), 200
+    else: 
+      return {'data': { 'id': 'at_risk_user', 'attributes': {'error': "User not found"} } }, 400
```
Comment (adumortier) [inline trackbasket_be/resources/at_risk_user.py:12]: thanks for adding the status code

File: trackbasket_be/resources/at_risk_user.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  
+  def post(self, id):
+    data = request.json 
+    at_risk_user = AtRiskUser(at_risk_user_id=id, **data)
```
[the remark this answers is not in the document]
Comment (adumortier) [reply trackbasket_be/resources/at_risk_user.py:16]: nice use of the kargs

**Pull request merged**